One-line show for SparseVector

jishnub opened the pull request:
Fix https://github.com/JuliaSparse/SparseArrays.jl/issues/400 by implementing the suggestion in https://github.com/JuliaSparse/SparseArrays.jl/issues/400#issuecomment-1640196626. 

After this,
```julia
julia> S = sparsevec([1,4], [2,3])
4-element SparseVector{Int64, Int64} with 2 stored entries:
  [1]  =  2
  [4]  =  3

julia> show(S)
sparsevec([1, 4], [2, 3], 4)
julia> [S,S]
2-element Vector{SparseVector{Int64, Int64}}:
 sparsevec([1, 4], [2, 3], 4)
 sparsevec([1, 4], [2, 3], 4)
```

The displayed form is round-trippable as well.

matbesancon commented:
@jishnub fantastic! Some docstrings need fixing now that the printing changed: https://github.com/JuliaSparse/SparseArrays.jl/actions/runs/5985741971/job/16238147916?pr=427

Also, we should probably add some tests beyond the `undef` one you changed

jishnub commented:
Updated, and added tests. I think this works now. This doesn't change the display for arrays with zero elements:
```julia
julia> S = sparsevec(Int64[], Int64[], 3)
3-element SparseVector{Int64, Int64} with 0 stored entries

julia> [S]
1-element Vector{SparseVector{Int64, Int64}}:
 3-element SparseVector{Int64, Int64} with 0 stored entries

julia> show(S)
sparsevec(Int64[], Int64[], 3)
```
The difference arises as the display of `S` has only one line even in "text/plain", which makes `[S]` use it directly instead of using `show(S)` as in non-empty cases. I'm unsure what's the best way to address this, so I've left this unchanged for now (and this doesn't matter for the original issue)

matbesancon commented on docs/Project.toml:
```diff
@@ -1,2 +1,5 @@
 [deps]
 Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
+
+[compat]
```
is that part necessary? This will create unnecessary conflicts with https://github.com/JuliaSparse/SparseArrays.jl/pull/444

jishnub replied:
This is necessary to get the docs to build until the upgrade to Documenter v1 is complete

ViralBShah replied:
I just merged #444.

matbesancon commented:
> The difference arises as the display of S has only one line even in "text/plain", which makes [S] use it directly instead of using show(S) as in non-empty cases. I'm unsure what's the best way to address this, so I've left this unchanged for now (and this doesn't matter for the original issue)

@jishnub I believe it would make sense to also cover this case but this can be done in a separate issue, can you create one?

matbesancon commented:
In the meantime, we can merge this, thanks a lot!

dkarrasch commented:
@matbesancon Please be careful about the merge option. This should have better been squashed and merged. For next time. 😉

matbesancon commented:
Oh my bad! I usually do, it slipped here
